### PR TITLE
Example codes of parallel read/write with independent/collective IO

### PR DESCRIPTION
**Pull request by valiantljk**

### CODE REVIEW

File: examples/create_file.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import h5py 
+f=h5py.File('test.h5','w')
+import numpy as np
+arr = np.arange(100)
+dset=f.create_dataset('dset',(10,10),data=arr)
+f.close()
+
```
Comment (aragilar) [inline examples/create_file.py]: This file doesn't seem that useful, given creating files is already covered in the docs.

File: examples/parallel_read.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+#open the dataset, get the handle, shape
+dset = f[dataset]
+
+
+length_x = dset.shape[0]
+length_y = dset.shape[1]
+if rank==0:
+   print (length_x,length_y)
+
+f.atomic = False # for better performance
```
Comment (aragilar) [inline examples/parallel_read.py]: Can you go into more details here?

**Comment by aragilar** [comment]
Can you make the following changes:
- [ ] Make the examples python 2/3 compatible
- [ ] Use with statements
- [ ] Cleanup the style (e.g. make flake8 clean)
- [ ] Don't make all the files executable
- [ ] Rebase on master to remove merge commits

File: examples/parallel_read.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+#devide the workload,
```
Comment (aragilar) [inline examples/parallel_read.py]: typo devide → divide

File: examples/create_file.py (hunk 1 above)
Comment (valiantljk) [inline examples/create_file.py]: agree. Will remove this.

File: examples/parallel_read.py (hunk 2 above)
Comment (valiantljk) [inline examples/parallel_read.py]: Do you mean the f.atomic thing?

File: examples/parallel_read.py (hunk 3 above)
Comment (valiantljk) [inline examples/parallel_read.py]: Sounds good to me, I'll make the changes. Thanks @aragilar

File: examples/parallel_read.py (hunk 2 above)
Comment (aragilar) [inline examples/parallel_read.py]: Yep, presumably there are trade offs whether atomic access or not is used. We don't need a paragraph, just a sentence or 2, and maybe a link to the docs about atomic access.

File: examples/parallel_read.py (hunk 3 above)
Comment (valiantljk) [inline examples/parallel_read.py]: can you tell me how to rebase on master to remove the merge commits? I tried, but have no idea how to do that.

Comment (aragilar) [inline examples/parallel_read.py]: You want something like `git rebase --onto h5py/master start-branch end-branch`, where `h5py` is the name of the remote which points to this repo, `start-branch` is the point where you diverged from what's in this repos master, and `end-branch` is your current branch. If you can't get this working, I can clean this up, but if you can do the rest that'd be helpful.

**Comment by takluyver** [comment]
This work appears to have stalled. Is anyone interested in picking it up? You'd need to rebase, polish up the example code involved, and undo the permission changes so it doesn't touch all the files.

In the meantime, should we close the PR?

**Comment by takluyver** [comment]
This appears to be stalled; I'm going to close it to tidy up the list of open PRs a bit. @valiantljk, if you're interested in picking it up again, we'll happily reopen it - or you can create a new PR if you prefer.

**Pull request closed without merging**